internal/tiltfile/files.go

```diff
@@ -372,3 +373,84 @@ func (s *tiltfileState) listdir(thread *starlark.Thread, fn *starlark.Builtin, a
 
 	return starlark.NewList(ret), nil
 }
+
+func (s *tiltfileState) jsonDecode(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var jsonString starlark.String
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs, "json", &jsonString); err != nil {
+		return nil, err
+	}
+
+	var decodedJSON interface{}
+
+	if err := json.Unmarshal([]byte(jsonString), &decodedJSON); err != nil {
+		return nil, err
+	}
+
+	return convertJsonToStarlark(decodedJSON)
+}
+
+func (s *tiltfileState) readJson(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var path starlark.String
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs, "path", &path); err != nil {
+		return nil, err
+	}
+
+	localPath, err := s.localPathFromSkylarkValue(path)
+	if err != nil {
+		return nil, fmt.Errorf("Argument 0 (path): %v", err)
+	}
+
+	contents, err := s.readFile(localPath)
```

Comment:
I made a two-line Tiltfile:

```
read_json('foo.json')
print(x["foo"])
```
and changes to foo.json didn't trigger a Tiltfile build

but then if I made the same Tiltfile with `read_file`, I hit the same problem, so I guess it's separate from this PR.

Reply:
filed CH1839